superset/tasks/async_queries.py

```diff
@@ -106,10 +112,16 @@ def load_explore_json_into_cache(  # pylint: disable=too-many-locals
 ) -> None:
     cache_key_prefix = "ejr-"  # ejr: explore_json request
 
-    user = (
-        security_manager.get_user_by_id(job_metadata.get("user_id"))
-        or security_manager.get_anonymous_user()
-    )
+    if user_id := job_metadata.get("user_id"):
+        # logged in user
+        user = security_manager.get_user_by_id(user_id)
+    elif guest_token := job_metadata.get("guest_token"):
+        # embedded guest user
+        user = security_manager.get_guest_user_from_token(guest_token)
+        del job_metadata["guest_token"]
+    else:
+        # default to anonymous user if no user is found
+        user = security_manager.get_anonymous_user()
```

Comment:
it looks like we can dry up this logic to assign a user. I know there's been some discussions in the past about this, maybe @jfrag1 or @john-bodley may have some opinions, but I think we could even move this to the security manager. I see a lot of different ways that we can fetch users like this across the app, but none that combine them like this. It's possible that this user definition could be used in other places.

Reply:
I feel like setting the user context here is still async queries domain knowledge so a bit hesitate to generalize it to security_manager

Reply:
Ok, this makes sense.